src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -0,0 +1,236 @@
+                    handleApplyUpdateRequest:commandParams
+                           completionHandler:^(MTROtaSoftwareUpdateProviderClusterApplyUpdateResponseParams * _Nullable data,
+                               NSError * _Nullable error) {
+                               chip::app::Clusters::OtaSoftwareUpdateProvider::Commands::ApplyUpdateResponse::Type response;
+                               ConvertFromApplyUpdateRequestResponseParms(data, response);
+
+                               chip::app::CommandHandler * handler = handle.Get();
+                               if (handler) {
+                                   handler->AddResponse(cachedCommandPath, response);
+                                   handle.Release();
+                               }
+                           }];
+            });
+        }
+    }
+}
+
+void MTROTAProviderDelegateBridge::HandleNotifyUpdateApplied(chip::app::CommandHandler * commandObj,
+    const chip::app::ConcreteCommandPath & commandPath,
+    const chip::app::Clusters::OtaSoftwareUpdateProvider::Commands::NotifyUpdateApplied::DecodableType & commandData)
+{
+    // Make sure to hold on to the command handler and command path to be used in the completion block
+    __block chip::app::CommandHandler::Handle handle(commandObj);
+    __block chip::app::ConcreteCommandPath cachedCommandPath(
+        commandPath.mEndpointId, commandPath.mClusterId, commandPath.mCommandId);
+
+    id<MTROTAProviderDelegate> strongDelegate = mDelegate;
+    if ([strongDelegate respondsToSelector:@selector(handleNotifyUpdateApplied:completionHandler:)]) {
+        if (strongDelegate && mQueue) {
+            auto * commandParams = [[MTROtaSoftwareUpdateProviderClusterNotifyUpdateAppliedParams alloc] init];
+            ConvertToNotifyUpdateAppliedParams(commandData, commandParams);
+
+            dispatch_async(mQueue, ^{
+                [strongDelegate
+                    handleNotifyUpdateApplied:commandParams
+                            completionHandler:^(NSError * _Nullable error) {
+                                chip::app::CommandHandler * handler = handle.Get();
+                                if (handler) {
+                                    handler->AddStatus(cachedCommandPath, chip::Protocols::InteractionModel::Status::Success);
+                                    handle.Release();
+                                }
+                            }];
+            });
+        }
+    }
+}
+
+void MTROTAProviderDelegateBridge::ConvertToQueryImageParams(
+    const chip::app::Clusters::OtaSoftwareUpdateProvider::Commands::QueryImage::DecodableType & commandData,
+    MTROtaSoftwareUpdateProviderClusterQueryImageParams * commandParams)
+{
+    commandParams.vendorId = [NSNumber numberWithInt:commandData.vendorId];
+    commandParams.productId = [NSNumber numberWithInt:commandData.productId];
+    commandParams.softwareVersion = [NSNumber numberWithInt:commandData.softwareVersion];
```

Comment:
And this `numberWithUnsignedLong`?  Otherwise we can get data corruption here if the uint32_t is outside the int range....